packages/a11y/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types",
+		"tsBuildInfoFile": "../../.cache/a11y.tsbuildinfo"
+	},
+	"references": [ { "path": "../dom-ready" } ],
```

Comment:
I see some mention of project references in the original comment, but it's not really clear to me what purpose it serves, or how I as the implementing developer should know what to include and when. Is it something we could document?

Reply:
In short, references should reflect any `@wordpress/*` (Gutenberg monorepo) dependencies for the package that also have types generated via `tsc`.

---

Do you have a good idea where it should go in the docs?

The TypeScript docs on [_Project References_ can be found here](https://www.typescriptlang.org/docs/handbook/project-references.html).

> Project references are a new feature in TypeScript 3.0 that allow you to structure your TypeScript programs into smaller pieces.
>
> By doing this, you can greatly improve build times, enforce logical separation between components, and organize your code in new and better ways.

This is analogous to packages the Gutenberg monorepo. The interesting bit is that `tsc` will take care of resolving an rebuilding referenced projects as required:

> Running tsc --build (tsc -b for short) will do the following:
> - Find all referenced projects
> - Detect if they are up-to-date
> - Build out-of-date projects in the correct order

Reply:
There's some prior art for documentation of packages development workflows at [`packages/README.md`](https://github.com/WordPress/gutenberg/blob/master/packages/README.md). Maybe a section there about TypeScript and/or `tsconfig.json` specifically?

Reply:
I've [added documentation](https://github.com/WordPress/gutenberg/pull/18942/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.yml&short_path=353fb41#diff-353fb419a3a39a1b358ab0ff9cd560de).